client/modules/core/main.js

```diff
@@ -387,25 +388,35 @@ export default {
   },
 
   getUserPreferences(packageName, preference, defaultValue) {
-    const user = Meteor.user();
-
-    if (user) {
-      const { profile } = Meteor.user();
-      if (profile && profile.preferences && profile.preferences[packageName] && profile.preferences[packageName][preference]) {
-        return profile.preferences[packageName][preference];
+    getDep(`${packageName}.${preference}`).depend();
+    if (Meteor.user()) {
+      const packageSettings = store.get(packageName);
+      // packageSettings[preference] should not be undefined or null.
+      if (packageSettings && typeof packageSettings[preference] !== "undefined" && packageSettings[preference] !== null) {
+        return packageSettings[preference];
       }
     }
 
     return defaultValue || undefined;
   },
 
   setUserPreferences(packageName, preference, value) {
+    getDep(`${packageName}.${preference}`).changed();
+    // User preferences are not stored in Meteor.user().profile
+    // to prevent all autorun() with dependency on Meteor.user() to run again.
     if (Meteor.user()) {
-      return Meteor.users.update(Meteor.userId(), {
-        $set: {
-          [`profile.preferences.${packageName}.${preference}`]: value
-        }
-      });
+      // "reaction" package settings should be synced to
+      // the Accounts collection.
+      if (packageName in ["reaction"]) {
```

Comment:
So if we only sync Reaction package, how will plugin-creators packages differ? Will it not be reactive when their settings change or will it not save the state of their panel? What's the thinking behind segregating them like this?

Reply:
> So if we only sync Reaction package, how will plugin-creators packages differ

I didn't understand this, maybe because I am not very clear how the plugin-creators packages work.
The reason for segregation here is that the `activeShopId` is stored in the `reaction` package, this is also used on the server side([here](https://github.com/reactioncommerce/reaction/blob/master/server/api/core/core.js#L379)).
So only if the `reaction` package is synced to backend, `activeShopId` would be the correct one.